core/src/optimistic_confirmation_verifier.rs

```diff
@@ -85,6 +85,9 @@ impl OptimisticConfirmationVerifier {
                     );
                 }
                 datapoint_info!("optimistic_slot", ("slot", new_optimistic_slot, i64),);
+                if let Some(metrics) = agave_metrics::singleton_metrics::get_metrics() {
+                    metrics.core.optimistic_slot.set(new_optimistic_slot as i64);
+                }
```

Comment:
Singleton is convenient here, but we still need to add these 3 lines everywhere `datapoint` or `counter` is used. I was wondering if possible to add them _into_ `datapoint_info!` macro. So no change on call sites, all "metrics server dispatches" happens behind macros.

Reply:
I guess some extra code is needed to turn on/off prometheus easily. for the macro, I'm not sure could we do something similar. we will need to register datapoint first then we used https://github.com/anza-xyz/agave/pull/2658/files#diff-fcfe3478edcc31f079d606b9d022bd0d359e21687b996c7f9b580fd62d97a0bcR19-R23. let me think about it